### DISCUSSION TITLE
Required input in viewfield

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#3427] [Help]
How can I validate an input text that is required that is inside a ViewField? inside a Wizard Layout

**Comment by danharrin** [comment]
Send your field code

**Reply by jcgmalave** [reply]
<input wire:model.lazy="detail_price.{{$key}}" required="required" type="number" inputmode="decimal" step="any" >

**Reply by danharrin** [reply]
So what's the problem with that `required`?

**Reply by jcgmalave** [reply]
despite the fact that the fields that I have inside the viewfield have the require property, when going to the next step of the wizard, it does not validate if it is empty or not

**Reply by danharrin** [reply]
The `$getStatePath()` part of our custom field example is important, if you use it then your form component will be bound to the correct property, and when you use `required()` then it will show the validation error correctly.

`wire:model.lazy="{{$getStatePath()}}"`

**Reply by jcgmalave** [reply]
I understand it dan, what I don't see now if it works the same way when I generate an array with the fields, by concatenating a key value that I generate inside a foreach, for example: wire:model.lazy="{{$getStatePath() }}.{{$key}}"

**Reply by danharrin** [reply]
How are you adding this field to your form schema?

**Reply by jcgmalave** [reply]
I instantiate a ViewField input within the form schema, and in the .blade that I associate with it, make a query, depending on the result of that query, I generate as many inputs as it returns, for example if it returns 4 records, I generate 4 inputs; the operation is dynamic

**Reply by danharrin** [reply]
You should use Alpine for this, and properly bind the array of data to the `state` property. Then you can add a custom required() validation rule that loops through the array and ensures each input is present
